### PR TITLE
Narrow the scope of the ``unnecessary-ellipsis`` checker 

### PR DESCRIPTION
**Pull request by mbyrnepr2**
Narrow the scope of the ``unnecessary-ellipsis`` checker to functions & classes which contain both a docstring and an ellipsis.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
|    | :scroll: Docs          |

## Description

Closes #XXX

### CODE REVIEW

File: tests/functional/u/unnecessary/unnecessary_ellipsis.py [hunk 1]
```diff
@@ -9,10 +9,10 @@
     A = 2
 except ValueError:
     A = 24
-    ... # [unnecessary-ellipsis]
+    ...
```
Comment (DanielNoord) [inline tests/functional/u/unnecessary/unnecessary_ellipsis.py:12]: Hm, don't we think these are unnecessary?

Comment (mbyrnepr2) [reply tests/functional/u/unnecessary/unnecessary_ellipsis.py:12]: True. I simplified the checker to an extreme to prefer false negatives over false positives in order to make things as safe as possible to users. There will likely be other scenarios we can emit it also. We can allow this case in the current MR or in another one.

Comment (DanielNoord) [reply tests/functional/u/unnecessary/unnecessary_ellipsis.py:12]: Would `node.parent.parent.doc_node or len(node.parent.parent.body) > 1 work? Or something like that? Every `FunctionDef` or `ClassDef` that has a docstring or a body that has more than 1 node? That also reflects the description of the message a little better.

Comment (mbyrnepr2) [reply tests/functional/u/unnecessary/unnecessary_ellipsis.py]: That works well with the tests; lets try this. Thanks @DanielNoord.

**Review by DanielNoord — APPROVED** [review]
Perfect! Glad my suggestion worked!

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas Does this need an entry in `2.13.rst`?

**Review by Pierre-Sassoulas — APPROVED** [review]
It's ok, I think it's only a refactor.

**Pull request merged**